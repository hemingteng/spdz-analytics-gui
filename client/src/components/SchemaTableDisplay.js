/**
 * Panel to display a DB schema table.
 */
import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'
import { DisplayBox, codeFont } from './BaseStyles'

const DivDBTable = DisplayBox.extend`padding: 0px 2px 4px 2px;`
const TableSQL = styled.table`
  width: 100%;
  border-collapse: collapse;
  font-family: ${codeFont.fontFamily};
  font-size: ${codeFont.fontSize};
  color: ${codeFont.color};
`
const TableCaption = styled.caption`padding-left: 4px;`
const TdCol = styled.td`
  padding: 0 6px 0 6px;
  border: none;
  text-align: left;
`

const SchemaTableDisplay = props => {
  const formatTableCols = colNames => {
    return Object.keys(colNames).map(colName =>
      <tr key={colName}>
        <TdCol>
          {colName}
        </TdCol>
        <TdCol>
          {colNames[colName].type} {colNames[colName].maxLength}
        </TdCol>
      </tr>
    )
  }

  return (
    <DivDBTable>
      <TableSQL>
        <TableCaption>
          {props.tableName}
        </TableCaption>
        <tbody>
          {formatTableCols(props.colNames)}
        </tbody>
      </TableSQL>
    </DivDBTable>
  )
}

SchemaTableDisplay.propTypes = {
  tableName: PropTypes.string.isRequired,
  colNames: PropTypes.object.isRequired
}

export default SchemaTableDisplay
