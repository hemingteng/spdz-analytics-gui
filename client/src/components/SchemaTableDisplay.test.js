import React from 'react'
import renderer from 'react-test-renderer'
import SchemaTableDisplay from './SchemaTableDisplay'

const testCols = { 'columns': { 'id': { 'type': 'int', 'maxLength': null }, 'amount': { 'type': 'int', 'maxLength': null }, 'ipAddress': { 'type': 'int', 'maxLength': null }, 'incidentDate': { 'type': 'datetime', 'maxLength': null } } }


describe('Schema table display renders as expected', () => {
  it('Renders as expected', () => {
    const tree = renderer.create(
      <SchemaTableDisplay tableName={'testTable'} colNames={testCols.columns} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
