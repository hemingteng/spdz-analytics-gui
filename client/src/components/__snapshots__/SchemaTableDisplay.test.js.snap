// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema table display renders as expected Renders as expected 1`] = `
.c0 {
  background-color: #f5f5f5;
  border: 1px solid #ccc;
  border-radius: 4px;
  text-align: left;
  padding: 0px 2px 4px 2px;
}

.c1 {
  width: 100%;
  border-collapse: collapse;
  font-family: Menlo,Monaco,Consolas,"Courier New",monospace;
  font-size: 12.6px;
  color: #c7254e;
}

.c2 {
  padding-left: 4px;
}

.c3 {
  padding: 0 6px 0 6px;
  border: none;
  text-align: left;
}

<div
  className="c0"
>
  <table
    className="c1"
  >
    <caption
      className="c2"
    >
      testTable
    </caption>
    <tbody>
      <tr>
        <td
          className="c3"
        >
          id
        </td>
        <td
          className="c3"
        >
          int
           
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          amount
        </td>
        <td
          className="c3"
        >
          int
           
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          ipAddress
        </td>
        <td
          className="c3"
        >
          int
           
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          incidentDate
        </td>
        <td
          className="c3"
        >
          datetime
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
